src/fsharp/FSharp.Core/async.fs

```diff
@@ -1368,7 +1368,7 @@ namespace Microsoft.FSharp.Control
             let cancellationToken = defaultArg cancellationToken defaultCancellationTokenSource.Token
             AsyncPrimitives.StartWithContinuations cancellationToken computation id (fun edi -> edi.ThrowAny()) ignore
 
-        static member Sleep millisecondsDueTime : Async<unit> =
+        static member Sleep (millisecondsDueTime: int64) : Async<unit> =
```

Comment:
Since you added the `int64` overload here, would you mind pulling this into the interface file?

Reply:
May also require `[<Experimental(ExperimentalAttributeMessages.RequiresPreview)>]`

Reply:
I intentionally kept the int64 overload private. Are you sure we should add it to the public interface? Then we'd have int(32), int64 and TimeSpan overloads. If someone wants to sleep for more than 24.8 days (which is the maximum possible with int32), he'd be better off using the TimeSpan overload anyway - Async.Sleep(25. |> TimeSpan.FromDays) is much safer than Async.Sleep(1000L*60L*60L*24L*25L). Let me know if I should make it public or keep it private!

Reply:
Private is probably fine. Yeah, a sleep for that long would be pretty ridiculous :)